src/components/views/messages/DateSeparator.tsx

```diff
@@ -36,18 +38,25 @@ function getDaysArray(): string[] {
 interface IProps {
     ts: number;
     forExport?: boolean;
+    now?: Date;
 }
 
 @replaceableComponent("views.messages.DateSeparator")
 export default class DateSeparator extends React.Component<IProps> {
+    // use defaultProp for now to make testing easier
+    public static defaultProps = {
+        now: new Date(),
+    };
+
     private getLabel() {
         const date = new Date(this.props.ts);
+        const disableRelativeTimestamps = !SettingsStore.getValue(UIFeature.TimelineEnableRelativeDates);
 
         // During the time the archive is being viewed, a specific day might not make sense, so we return the full date
-        if (this.props.forExport) return formatFullDateNoTime(date);
+        if (this.props.forExport || disableRelativeTimestamps) return formatFullDateNoTime(date);
 
-        const today = new Date();
-        const yesterday = new Date();
+        const today = this.props.now;
```

Comment:
I agree it's not the nicest, but mocking the date constructor is fussy because the component uses `new Date()` for now and to transform the `ts` prop into a date for comparison.

Reply:
Sure, but both return the same date which is then mutated. Just mock the constructor and return a real Date object based on the same ts